README.md

<!-- 
README.md is generated from README.Rmd. 
Please edit that file 
-->

# npfda: Nonparametric functional data analysis

### Version 0.1.4

This package implements nonparametric methods for inference on
(multidimensional) functional data using the tools available in package
[`npsp`](https://rubenfcasal.github.io/npsp).

## Main functions

- `npf.data()` defines a (multidimensional) functional data set.

Nonparametric methods for inference on the functional trend, the
functional variance and the variogram:

- `npf.fit()` (automatically) fits a nonparametric functional model by
  estimating the trend, the conditional variance and the variogram.

- `locpol()`, `np.var()` and `np.svar()` methods use local polynomial
  kernel smoothing to compute nonparametric estimates of the functional
  trend, the functional variance and the (intra-curve) variogram (or
  their first derivatives), respectively.

See the
[Reference](https://rubenfcasal.github.io/npfda/reference/index.html)
for the complete list of functions.

## Installation

`npfda` is not available from CRAN, but you can install the development
version from github with:

``` r
# install.packages("remotes")
remotes::install_github("rubenfcasal/npfda")
```

<!-- 
Note also that, as this package requires compilation, Windows users need to have previously installed the appropriate version of [Rtools](https://cran.r-project.org/bin/windows/Rtools/), and OS X users need to have installed [Xcode](https://apps.apple.com/us/app/xcode/id497799835).

Alternatively, Windows users may install the corresponding *npfda_X.Y.Z.zip* file in the [releases section](https://github.com/rubenfcasal/npfda/releases/latest) of the github repository.

For R versions 4.2.x under Windows: 
``` r
install.packages('https://github.com/rubenfcasal/npfda/releases/download/v0.1-1/npfda_0.1-1.zip',
                 repos = NULL)
``` 
-->

## Authors

- [Rubén Fernández-Casal](https://rubenfcasal.github.io) (Dep.
  Mathematics, University of A Coruña, Spain).

- Sergio Castillo-Páez (Universidad de las Fuerzas Armadas ESPE,
  Ecuador).

- Miguel Flores (Faculty of Administrative Sciences, Escuela Politécnica
  Nacional, Ecuador).

Please send comments, error reports or suggestions to
<rubenfcasal@gmail.com>.

## Acknowledgments

This research has been supported by MICINN (Grant PID2020-113578RB-I00).
The research of Rubén Fernández-Casal has been supported by the Xunta de
Galicia (Grupos de Referencia Competitiva ED431C-2020-14 and Centro de
Investigación del Sistema Universitario de Galicia ED431G 2019/01). All
these grants were co-financed by the ERDF. The research of Sergio
Castillo Páez has been supported by the Universidad de las Fuerzas
Armadas ESPE, from Ecuador.

## References

- Fernández-Casal R. (2023) npsp: Nonparametric spatial (geo)statistics.
  R package version 0.7-11, *<https://rubenfcasal.github.io/npsp>*.

- Fernández-Casal R., Castillo-Páez S. and García-Soidán P. (2017),
  Nonparametric estimation of the small-scale variability of
  heteroscedastic spatial processes, *Spa. Sta.*, **22**, 358-370,
  [DOI](https://doi.org/10.1016/j.spasta.2017.04.001).

- Shapiro A. and Botha J.D. (1991) Variogram fitting with a general
  class of conditionally non-negative definite functions. *Computational
  Statistics and Data Analysis*, **11**, 87-96.


NEWS.md
# npfda 0.1-4 (2023-05-15) 

* Added S3 generic `npf.fit()` (with methods for `npf.data` 
  and `npf.locpol` classes).
  
* Updated package vignette.  

* Added pkgdown function reference index.  


# npfda 0.1-3 (2023-05-12) 

* Added `h.cv()` S3 methods for `npf.bin` and `npf.bin.res2` classes.

* Added `npf.model()` (constructor of `npf.model`-class) 

* Added `residuals()` S3 methods for `npf.locpol` and `npf.model` classes.

* Added `plot()` S3 methods for `npf.locpol` and `npf.model` classes.
  

# npfda 0.1-2 (2023-05-10) 

* Added `svar.bin()` S3 methods for `npf.data` and 
  `npf.locpol` classes (results `data` components are not fully compatible 
  with the `npsp` package).

* Added `np.svar()` S3 method for `npf.data` class.


# npfda 0.1-1 (2023-05-07) 

* Added `npf.bin.res2()` and S3 generic `np.var()` (with methods for `npf.locpol` 
  and `npf.bin.res2` classes; results `data` components are not fully compatible 
  with the `npsp` package). 
  
* Added `predict()` S3 methods for `npf.locpol` and `npf.var` classes.

* Changes in the `data` component of the return value of `npf.binning()`
  (results are now not fully compatible with the `npsp` package;
  care should be taken when using this function together with `npsp::as.bin.data()`).

* Classes `npf.locpol.bin` and `npf.bin.data` renamed as `npf.locpol` and `npf.bin`, 
  respectively.

  
# npfda 0.1-0 (2023-05-01) 

* Initial version in package form.
  


R/npfda_package.R
#····································································
#   npfda_package.R
#····································································
#
#   Authors: Ruben Fernandez-Casal, Miguel Flores
#   Created: May 2023
#
#   NOTE: Press Ctrl + Shift + O to show document outline in RStudio
#····································································
# pkgdown::build_site(getwd())

#····································································
# npfda-package ----
#····································································
#' npfda: Nonparametric functional data analysis
#'
#' This package implements nonparametric methods for inference
#' on (multidimensional) functional data using the tools available in package
#' [`npsp`](https://rubenfcasal.github.io/npsp).
#'
#' For more information visit <https://rubenfcasal.github.io/npfda.html>.
#' @keywords nonparametric spatial bootstrap Monte-Carlo
#' @name npfda-package
#' @aliases npfda
#' @docType package
#' @import npsp
#' @import graphics
#' @import stats
#' @importFrom grDevices rainbow
#' @references
#' Fernández-Casal R. (2023) npsp: Nonparametric spatial (geo)statistics.
#' R package version 0.7-11, <https://rubenfcasal.github.io/npsp>.
#'
#' Fernández-Casal R., Castillo-Páez S. and García-Soidán P. (2017), Nonparametric
#' estimation of the small-scale variability of heteroscedastic spatial processes,
#' *Spa. Sta.*, **22**, 358-370, [DOI](https://doi.org/10.1016/j.spasta.2017.04.001).
NULL


#····································································
# ozone ----
#····································································
#' Ground-level ozone
#'
#' Daily averages of ozone concentration (microgram per cubic meter) recorded over
#' the period from 1988 to 2020 at the Yarner Wood [AURN](https://uk-air.defra.gov.uk/networks/network-info?view=aurn)
#' monitoring site in the UK.
#'
#' It is assumed that the observations corresponding to each year are (partial)
#' realizations of a functional process, so the data consist of 33 curves observed
#' at 365 discretization points.
#' @source
#' The UK Automatic Urban and Rural Network ([AURN](https://uk-air.defra.gov.uk/networks/network-info?view=aurn)).
#' The data were downloaded from \url{https://uk-air.defra.gov.uk/data} and preprocessed
#' with the [`climatol`](https://climatol.eu/) package (by applying the usual outlier
#' detection and data imputation methods).
#' @name ozone
#' @docType data
#' @format A matrix with 365 rows (days) and 33 columns (years).
#' @keywords datasets
#' @examples
#' fd.ozone <- npf.data(ozone, dimnames = "day")
#' plot(fd.ozone)
NULL
# load("ozono.RData")
# ozone <- as.matrix(ozono[-1])
# colnames(ozone) <- 1988:2020
# rownames(ozone) <- 1:365
# usethis::use_data(ozone)


#--------------------------------------------------------------------
.onAttach <- function(libname, pkgname){
  #--------------------------------------------------------------------
  #   pkg.info <- utils::packageDescription(pkgname, libname, fields = c("Title", "Version", "Date"))
  pkg.info <- drop( read.dcf( file = system.file("DESCRIPTION", package = "npfda"),
                              fields = c("Title", "Version", "Date") ))
  packageStartupMessage(
    paste0(" npfda: ", pkg.info["Title"], ",\n"),
    paste0(" version ", pkg.info["Version"], " (built on ", pkg.info["Date"], ").\n"))
}
